### ISSUE TITLE
User pages/bios

### ISSUE THREAD

**Issue by rachelrosenfelt** [misfist/tni-core-functionality#21]
Is it possible to have our about page include the complete list of contributors and link to their bios as we had on the old site? 
![screenshot 2017-05-05 14 12 18](https://cloud.githubusercontent.com/assets/14044305/25758544/eb3c8f78-319c-11e7-99a8-7135539841c8.png)

**Comment by misfist** [comment]
Yes, this is possible. The question is really who is considered a contributor? Is it anyone who is attributed as an author of a post or blog? Or is it specific users that are marked as contributors?

**Comment by rachelrosenfelt** [comment]
I'm not sure how the database is structured, but in the old site, if we checked the box that they were a contributor (as opposed to their user role selected from the pulldown menu, which had a contributor option too, but could indicate anything). If the fact of that box getting checked exists in our database, that's what I'd pull from.

**Comment by misfist** [comment]
We can't use user roles for this because, since we're now using Co-authors Plus, not all authors will have a user account. But, if we don't want to show all authors, then there will need to be a way of indicating which should be displayed on contributor page. So, we could either add a way to pick a role (e.g. contributor, editor, blogger, etc.) for each author or check off if the author should appear in contributor list (e.g. "Display in contributor list").

**Comment by frnsys** [comment]
hm picking a role is probably the most flexible for potential future needs, so I'd say we should go with that

**Comment by misfist** [comment]
Yes, I agree.

**Comment by misfist** [comment]
This is ready for you to test on staging. 
http://tni-staging.positiondevapp.com/contributors/

You can assign a role to guest authors. The role choices are the same as whatever roles exist for WP users. In order to initially populate the guest author roles, I wrote and ran a utility function that automatically adds a role to each guest author, based on the linked user account's role (if there is a linked account).

To display the guest authors:

A listing can be added to a page using a shortcode `[guest-author-list]` or using the template tag `tni_core_coauthors_wp_list_authors()` in a template file.

I'm guessing you'll want to change the layout and styling.

**Comment by misfist** [comment]
Please note that Rachel mentioned the contributor names should got to a bio page, but we don't have a bio page. Instead, we have a author post listing. 

Example: http://tni-staging.positiondevapp.com/author/aaron-bady/

If a bio page is something you want, we could consider adding author name as page title and including their bio on the page.

**Comment by frnsys** [comment]
thanks for putting this together - yes, let's add the author name and bio at the top of their contributor page. Then I'll style everything. thanks!

**Comment by misfist** [comment]
The name and bio have been added, and the changes are now live.
